core/src/Streamly/Internal/Data/MutArray.hs

```diff
@@ -83,10 +87,100 @@ getSlicesFromLen :: forall m a. (Monad m, Unbox a)
     -> Unfold m (MutArray a) (MutArray a)
 getSlicesFromLen = slicerFromLen
 
--- | Scan @compactChunksLE n@ coalesces adjacent arrays in the input stream
+-- | Scan @compactLE n@ coalesces adjacent arrays in the input stream
 -- only if the combined size would be less than or equal to n.
-{-# INLINE compactChunksLE #-}
-compactChunksLE :: (MonadIO m, Unbox a) =>
+{-# INLINE compactLE #-}
+compactLE :: (MonadIO m, Unbox a) =>
     Int -> Stream m (MutArray a) -> Stream m (MutArray a)
--- compactLE n = Stream.catRights . Stream.parseManyD (pCompactChunksLE n)
-compactChunksLE = rCompactChunksLE
+-- XXX compactLE can be moved to MutArray/Type if we are not using the parser
+-- to implement it.
+-- compactLE n = Stream.catRights . Stream.parseManyD (pCompactLE n)
+compactLE = rCompactLE
```

Comment:
Interesting. Why do we do this? Why cant we rename `rCompactLE` to `compactLE`?

Reply:
`rCompactLE` is defined in `MutArray/Type`, it is a local function. compactLE is the one that can be exposed and it can be defined in an alternate way as in the comment above, that's why it has been defined in this module because it requires higher level stream module imports causing cyclic deps in Type.hs.